src/pages/Future/Index.js

```diff
@@ -9,34 +11,70 @@ import { getTokens, getContract } from '@/constants/powers.js';
 import AcySymbolNav from '@/components/AcySymbolNav';
 import AcySymbol from '@/components/AcySymbol';
 import { getGlobalTokenList } from '@/constants';
+import { fetcher } from '@/acy-dex-futures/utils';
+import Reader from '@/abis/future-option-power/Reader.json'
 import styles from './styles.less'
 
+
 const Future = props => {
-  const { chainId } = useChainId();
+  const { account, library, active } = useWeb3React();
+  let { chainId } = useChainId();
   const tokens = getTokens(chainId);
+  chainId = 80001
 
   const [mode, setMode] = useState('Buy')
   const [symbol, setSymbol] = useState('BTCUSD')
 
-  const [activeToken, setActiveToken] = useState((tokens.filter(ele => ele.symbol == "BTC"))[0]);
+  // const [activeToken, setActiveToken] = useState((tokens.filter(ele => ele.symbol == "BTC"))[0]);
+
+  const readerAddress = getContract(chainId, "reader")
+  const poolAddress = getContract(chainId, "pool")
+
+  const { data: symbolsInfo, mutate: updateSymbolsInfo } = useSWR([chainId, readerAddress, "getSymbolsInfo", poolAddress, []], {
+    fetcher: fetcher(library, Reader)
+  });
+  //future_tokens store every symbols in future and its data 
+  const future_tokens = symbolsInfo?.filter(ele=>ele[0] == "futures")
```

Comment:
This chunk can be moved into a `useMemo()` to reduce the overhead of recalculation